src/OpenTelemetry/Internal/CircularBuffer.cs

```diff
@@ -122,6 +122,57 @@ public bool TryAdd(T value)
             }
         }
 
+        /// <summary>
+        /// Attempts to add the specified item to the buffer.
+        /// </summary>
+        /// <param name="value">The value to add.</param>
+        /// <param name="maxSpinCount">The maximum allowed spin count, when set to a negative number of zero, will spin indefinitely.</param>
```

Comment:
We might want to avoid spin entirely if we are running on a single core machine. I guess it is a low priority at this moment.